mcs/class/System.Net.Http/MonoWebRequestHandler.cs

```diff
@@ -445,7 +446,17 @@ public async Task<HttpResponseMessage> SendAsync (HttpRequestMessage request, Ca
 		{
 			if (disposed)
 				throw new ObjectDisposedException (GetType ().ToString ());
-
+			
+			// We need to preserve the timeout provided in the cancellationToken if there is one. Unfortunately reflection is the only way to access this (case 1365107)
+			FieldInfo prop = typeof(CancellationToken).GetField ("_source", BindingFlags.NonPublic | BindingFlags.Instance | BindingFlags.Static);
```

Comment:
Do we have to worry about any of these fields being linked away? @mrvoorhe

Reply:
It looks like there may already be InternalsVisibleTo that we need, but we'd need to adjust some field visiblity? https://github.com/Unity-Technologies/mono/blob/unity-master/mcs/class/corlib/Assembly/AssemblyInfo.cs#L104

Not sure if it's better, just a thought.

Reply:
Right yeah if we attempted the `InternalsVisibleTo` trick we'd have to modify a bunch of fields to make them accessible. I guess the difference in implementation then would be a lot of touched files and a mild change to MonoWebRequestHandler.cs whereas this is pretty self contained but is leaning on reflection.

Reply:
There wasn't a test for this exact binding flags combo, but I added one and it's green.  The linker can correctly track all 3 of these GetField usages